solanches/authenticate.py

```diff
@@ -29,12 +26,16 @@ def wrapper(*args, **kwargs):
 
         _assert(token and not token_in_block_list, "Error: Você não tem permissão para acessar essa rota.", SolanchesNotAuthorizedError)
         
+        comercio_nome = kwargs.get("comercio_nome")
+
         try:
             decoded = jwt.decode(token, current_app.config['SECRET_KEY'], algorithms=["HS256"])
-            current_user = controller.get_comercio_by_id(decoded.get("id"))
+            current_user = controller.get_comercio_by_id(decoded.get("id")).get("nome")
         except:
             _assert(False, "Error: Token inválido ou expirado.", SolanchesNotAuthorizedError)    
+
+        _assert(not comercio_nome or current_user == comercio_nome, "Error: Token não referente a esse usuário.", SolanchesNotAuthorizedError)
```

Comment:
Esse `not comercio_nome` não permitiria que, sei lá, um `comercio_nome = None` consiga se autenticar? Por nossa lógica isso nunca aconteceria porque não tem como `comercio_nome` ser `None`, mas me parece um bad smell pro futuro. Verificar só se o `current_user == comercio_nome` não seria o suficiente?
```suggestion
        _assert(current_user == comercio_nome, "Error: Token não referente a esse usuário.", SolanchesNotAuthorizedError)
```

Reply:
Sim, isso me deixou com suspeitas tbm, mas no método de logout a gente tem um erro pq não recebe um comercio_nome. Aí pensei em usar essa solução ou mudar a rota de logout para /logout/<comercio_nome>. O que achas ?